generate-report.sh

```
@@ -43,18 +61,64 @@ echo "Month name: $MONTH_NAME"
 # Function to get the first name of a GitHub user
 get_first_name() {
   local github_handle=$1
+
+  # Check if the first name is already cached
+  if [[ -n "${FIRST_NAME_CACHE[$github_handle]}" ]]; then
+    echo "${FIRST_NAME_CACHE[$github_handle]}"
+    return
+  fi
+
+  # Fetch the full name from the GitHub API
   local full_name=$(gh api users/$github_handle --jq .name)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)